File: sanghoon/programmers-2/조이스틱.py
def solution(name):
    answer = 0
    min_move = len(name) - 1
    for i, char in enumerate(name) :
        up = ord(char) - ord('A')
        down = ord('Z') - ord(char) + 1
        answer += min(up, down)

        next = i + 1
        while next < len(name) and name[next] == 'A' :
            next += 1
        min_move = min([min_move, 2 *i + len(name) - next, i + 2 * (len(name) - next)])
    answer += min_move
    print(answer)
name = "JAN"
solution(name)

File: kyulee/programmers-2/조이스틱.java

public class 조이스틱 {
    public static int solution(int n) {
        int answer = 0;  // 총 조작 횟수
        int move = name.length() - 1;  // 좌우 이동 횟수 초기값 설정

        // 상하 이동 횟수 계산
        for (int i = 0; i < name.length(); i++) {
            // 현재 문자에 대한 상하 이동 횟수 추가
            answer += Math.min(name.charAt(i) - 'A', 'Z' - name.charAt(i) + 1);

            int index = i + 1;
            // 다음 A가 아닌 문자를 찾을 때까지 이동 횟수 계산
            while (index < name.length() && name.charAt(index) == 'A') {
                index++;
            }
            // 현재 위치에서의 좌우 이동 횟수 계산
            move = Math.min(move, i * 2 + name.length() - index);
            move = Math.min(move, (name.length() - index) * 2 + i);
        }

        // 최종 결과 반환
        return answer + move;
    }
}
